cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/CognitiveServiceBase.scala

```diff
@@ -91,7 +91,9 @@ trait HasServiceParams extends Params {
   }
 
   protected def shouldSkip(row: Row): Boolean = getRequiredParams.exists { p =>
-    emptyParamData(row, p)
+    if (emptyParamData(row, p))
+      throw new NullPointerException(s"required param undefined: $p")
```

Comment:
This will never throw false if the error is in here. What I'm thinking is that we look at the required params in the transformSchema function to ensure all required params are set with either a value or a column. We can then call transformSchema before we transform to add validation there too

Reply:
nit: Should be an IllegalArgumentException

Reply:
We could have the logic in getInternalTransformer. I think the logic should live in the "control-plane" which executes on the head node rather than code inside a mapPartitions like shouldSkip